camviewer_ui_impl.py

```diff
@@ -1151,30 +1151,58 @@ def shutdown(self):
 
     def onfileSave(self):
         try:
-            fileName = QFileDialog.getSaveFileName(
+            filename = QFileDialog.getSaveFileName(
                 self,
                 "Save Image...",
-                self.cwd,
+                os.path.expanduser("~"),
                 "Images (*.npy *.jpg *.png *.bmp *.pgm *.tif)",
             )[0]
-            if fileName == "":
+            if filename == "":
```

Comment:
It's not documented very well (https://doc.qt.io/qt-5/qfiledialog.html) but empty string is returned if the user clicks "cancel".

Reply:
No, I meant to respond here in this way, but maybe I didn't justify myself fully.

The intention is to exit if the user clicks "cancel", which is signaled by there being no filename. Otherwise the user has no way to change their mind and not save anything.

Reply:
Ah yeah, you're right- there's a UX problem here. I forgot what came next when responding to this comment since it was pre-existing code. I'll clean this up.

Reply:
Resolved by making this code path give no message at all. No raise, no pop-up- the UI element that gets us to this point doesn't let you input an empty string, so the only way to be here is to click "X" or "Cancel", so you don't need this confirmation pop-up in any form. Added a comment to clarify intent.